tsnake/initialize.py

```diff
@@ -0,0 +1,543 @@
+        
+        NOTE: we may eventually need to find a way to initialize 
+              the nodes a little more intelligently
+        
+        Args:
+        =====================================================
+        (int) N:
+        * The initial number of T-snake nodes.
+        
+        (float) p:
+        * The hyperparameter p from Equation (7).
+        
+        (float) c:
+        * The hyperparameter c from Equation (A.4).
+        
+        (float) sigma:
+        * The hyperparameter sigma from Equation (A.4).
+        
+        (float) a:
+        * The hyperparameter a from Equations (1), (8).
+        
+        (float) b:
+        * The hyperparameter b from Equations (1), (8).
+        
+        (float) gamma:
+        * The hyperparameter gamma from Equations (1), (8).
+        
+        (bool) verbose:
+        * Set to True to print extra information.
+        =====================================================
+        """
+        # =============================================================
+        # Argument checks =============================================
+        # =============================================================
+        assert N > 1, 'there must be at least 2 T-snake nodes (got N=%d).' % N
+        assert p > 0, 'Hyperparameter p (Eq 7) must be > 0 (got p=%f).' % p
+        assert c > 0, 'Hyperparameter c (Eq A.4) must be > 0 (got c=%f).' % c
+        assert a > 0, 'Hyperparameter a (Eqs 1,8) must be > 0 (got a=%f).' % a
+        assert b > 0, 'Hyperparameter b (Eqs 1,8) must be > 0 (got b=%f).' % b
+        assert dt > 0, 'Hyperparameter dt (Eq 8) must be > 0 (got dt=%f).' % dt
+        
+        msg = 'Hyperparameter sigma (Eq A.4) must be > 0 (got sigma=%f).' % sigma
+        assert sigma > 0, msg
+        msg = 'Hyperparameter gamma (Eqs 1,8) must be > 0 (got gamma=%f).' % gamma
+        assert gamma > 0, msg
+        # =============================================================
+        # =============================================================
+        
+        # pixels on the boundary between masked and unmasked
+        edge_pixels = self._find_edge_pixels()
+        
+        step = int(np.floor(len(edge_pixels) / N))
+        step = max(step, 1)
+        if verbose:
+            print('Initializing a Node at every %d-th boundary-pixel' % step)
+        
+        # Order the nodes so that they connect to closest nodes
+        ordered_nodes = self.order_snake_nodes(edge_pixels)
+        
+        # Pull out every step-th pixel for initializion as a Node
+        nodes = []
+        for node_num in range(0, len(ordered_nodes), step):
+            r, c = ordered_nodes[node_num]
+            new_node = Node(r, c)
+            nodes.append(new_node)
+            
+        if verbose:
+            print('Total of %d T-snake nodes were initialized' % len(nodes))
+        
+        force_grid = self.compute_force_grid(sigma, c)
```

Comment:
Hyperparameter p is never used. In my comment below, I used p in the function compute_force_grid

Reply:
Fixed